end_to_end/eval_assert.py

```diff
@@ -20,13 +21,32 @@ def read(metrics_file, target):
   return avg
 
 
+def test_tflops(metrics_file, target, threshold):
+  """Asserts over tflops values"""
+  avg_tflops = read(metrics_file, target)
+  assert avg_tflops >= threshold
+
+
+def test_checkpointing(metrics_file, target):
+  """Asserts over loss values from loaded checkpoint"""
+  metrics_file_saved = 'saved_' + metrics_file
+  metrics_file_restored = 'restored_' + metrics_file
+
+  with open(metrics_file_saved, 'r', encoding='utf8') as saved,\
+    open(metrics_file_restored, 'r', encoding='utf8') as restored:
+    saved_loss = json.loads(saved.readlines()[-1])[target]
+    restored_loss = json.loads(restored.readlines()[0])[target]
+    assert isclose(saved_loss, restored_loss, rel_tol=0.5)
```

Comment:
Can we include a comment what this assert is? Something like we are comparing that the first step of restore matches the last step of save (or maybe is one step difference) - for the purpose of testing that save/restoring works, not actually a performance assert.